torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -64,6 +63,10 @@ std::map<at::ScalarType, ncclDataType_t> ncclDataType = {
     {at::kLong, ncclInt64},
     {at::kHalf, ncclHalf},
     {at::kBool, ncclUint8},
+    {at::kFloat8_e5m2, ncclUint8},
+    {at::kFloat8_e4m3fn, ncclUint8},
+    {at::kFloat8_e4m3fnuz, ncclUint8},
+    {at::kFloat8_e5m2fnuz, ncclUint8},
```

Comment:
wondering what's the concerns of running reduction op on uint8 directly?

Reply:
I guess it depends on how nccl_reduce works, if only the comms are in uint8 but the actual computation is done in fp8 then should be fine. That beings said fp8 addition without a scale is not accurate so mabye this just isnt a problem

Reply:
I think the point is that nccl_reduction would perform both communication and reduction together when running allreduce or reduce_scatter. So if this is not accurate for fp8, then we should throw some error when hitting a reduction and a fp8 dtype, otherwise user would hit silent correctness issue

Reply:
Yeah okay this is what I thought would happen and is incorrect for fp8, ill add a torchcheck to the reduce ops